sdk/download.go

```diff
@@ -269,11 +269,12 @@ func UpdateFile(file, url string, client *http.Client) error {
 	}
 	t.Close()
 	tempFile := t.Name()
-	defer os.Remove(tempFile)
+	os.Remove(tempFile) // let DownloadFile create it and use umask
```

Comment:
The use of tempfiles is what causes the problem in the first place, so I don't see why more should be added.

If unlinking the file is a problem, maybe it could be solved by a manual calculation of an appropriate mode with umask and `chmod`ing the tmpfile instead.